src/commands/view.ts

```diff
@@ -0,0 +1,65 @@
+import * as vscode from "vscode";
+
+import { Command, CommandFlags, CommandState, InputKind, registerCommand } from ".";
+import { EditorState } from "../state/editor";
+
+const getMenu = (name: string) => (editorState: EditorState) => {
+  const menuItems = editorState.extension.menus.get(name)!.items;
+
+  return Object.entries(menuItems).map((x) => [x[0], x[1].text]) as [string, string][];
+};
+
+const executeMenuItem = async (editorState: EditorState, name: string, i: number) => {
+  const menuItems = editorState.extension.menus.get(name)!.items;
+  const menuItem = Object.values(menuItems)[i];
+
+  try {
+    await vscode.commands.executeCommand(menuItem.command, menuItem.args);
+  } catch (e) {
+    const str = `${e}`.replace(/^Error: /, "");
+
+    vscode.window.showErrorMessage(`Command did not succeed successfully: ${str}.`);
+  }
+};
+
+// TODO: Make just merely opening the menu not count as a command execution
+// and do not record it.
+registerCommand(
+  Command.view,
+  CommandFlags.ChangeSelections,
+  InputKind.ListOneItemOrCount,
+  getMenu("view"),
+  (editorState, state) => {
+    if (state.input !== null)
+      return executeMenuItem(editorState, "view", state.input);
+    return;
+  },
+);
+
+async function toCenter() {
+  let currentLineNumber = vscode.window.activeTextEditor?.selection.start.line;
```

Comment:
1. `let` -> `const`
2. Use the state given by `registerCommand`, instead of the global `vscode.window.activeTextEditor`. You can see how it's done in [`move.ts`](https://github.com/71/dance/blob/19658b1bf85fac9e98de2231cc5cca21b31de865/src/commands/move.ts#L103-L136).
3. Return the promise instead of using an async function.
4. Object properties should be followed by `,`.